### DISCUSSION TITLE
PaddleX/latest/module_usage/tutorials/cv_modules/instance_segmentation

### DISCUSSION THREAD

**Discussion by giscus** [PaddlePaddle/PaddleX#2613] [General]
# PaddleX/latest/module_usage/tutorials/cv_modules/instance_segmentation

https://paddlepaddle.github.io/PaddleX/latest/module_usage/tutorials/cv_modules/instance_segmentation.html

**Comment by chenkings-qq** [comment]
User-defined training process will produce training folders for each round, which takes up a lot of memory, but it's nothing to me. I just need to use weights to make predictions. How to solve this problem?

**Reply by cuicheng01** [reply]
You can try deleting them manually. Of course, we will optimize this part of the logic in the future, such as automatically deleting the weights saved from previous epochs.

**Comment by giswrok** [comment]
请问这个main.py 在哪里？

**Reply by cuicheng01** [reply]
需要clone代码，可以查看相关文档

**Comment by yocent** [comment]
python main.py -c paddlex/configs/modules/instance_segmentation/Mask-RT-DETR-L.yaml -o Global.mode=check_dataset -o Global.dataset_dir=./instance_seg_labelme_examples
Start loading json annotation files from instance_seg_labelme_examples\train_anno_list.txt ...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████| 746/746 [00:00<00:00, 2549.08it/s]
The converted annotations has been save to instance_seg_labelme_examples\annotations\instance_train.json.
Start loading json annotation files from instance_seg_labelme_examples\val_anno_list.txt ...
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████| 12/12 [00:00<00:00, 2400.06it/s]
The converted annotations has been save to instance_seg_labelme_examples\annotations\instance_val.json.
Convert dataset successfully !
Traceback (most recent call last):
  File "F:\S_Paddlex\paddlex\utils\result_saver.py", line 29, in wrap
    result = func(self, *args, **kwargs)
  File "F:\S_Paddlex\paddlex\engine.py", line 38, in run
    return self._model.check_dataset()
  File "F:\S_Paddlex\paddlex\model.py", line 97, in check_dataset
    return dataset_checker.check()
  File "F:\S_Paddlex\paddlex\modules\base\dataset_checker\dataset_checker.py", line 77, in check
    self.split_dataset(dataset_dir)
  File "F:\S_Paddlex\paddlex\modules\instance_segmentation\dataset_checker\__init__.py", line 66, in split_dataset
    return split_dataset(
  File "F:\S_Paddlex\paddlex\modules\instance_segmentation\dataset_checker\dataset_src\split_dataset.py", line 28, in split_dataset
    assert (
AssertionError: The sum of train_rate(0.9), val_rate(0.1) should equal 100!

**Reply by yocent** [reply]
这个报错啥意思

**Reply by yocent** [reply]
CheckDataset:
  convert: 
    enable: True
    src_dataset_type: LabelMe
  split: 
    enable: True
    train_percent: 0.9
    val_percent: 0.1

**Reply by yellowducklet** [reply]
split.train_percent	int	设置训练集的百分比，类型为0-100之间的任意整数，需要保证和val_percent值加和为100；	
split.val_percent	int	设置验证集的百分比，类型为0-100之间的任意整数，需要保证和train_percent值加和为100。

**Comment by armjj** [comment]
根据上面的流程进行到训练这一步的时候，报如下错误，应该怎么处理？
paddlex.utils.errors.others.UnsupportedParamError: 'Mask-RT-DETR-L' is not a registered model name.
